src/sinol_make/commands/init/__init__.py

```diff
@@ -80,13 +106,17 @@ def run(self, args: argparse.Namespace):
                 util.exit_with_error(f"Destination {destination} already exists. "
                                      f"Provide a different task id or directory name, "
                                      f"or use the --force flag to overwrite.")
-        os.chdir(destination)
+        try:
+            self.template_dir = self.download_template(args.template)
 
-        self.template_dir = self.download_template()
+            os.chdir(destination)
 
-        self.move_folder()
-        self.update_config()
+            self.move_folder()
+            self.update_task_id()
 
-        self.used_tmpdir.cleanup()
+            self.used_tmpdir.cleanup()
 
-        print(util.info(f'Successfully created task "{self.task_id}"'))
+            print(util.info(f'Successfully created task "{self.task_id}"'))
+        except:
+            shutil.rmtree(destination)
```

Comment:
Shouldn’t `self.used_tmpdir` also be cleaned up?

Reply:
It should be, thanks
Moved that cleanup into finally block